registry/registry.go

```diff
@@ -165,7 +165,18 @@ const (
 	Alt
 )
 
-var typeStrings = [...]string{"int", "number", "string", "bool", "map", "array", "any", "alt"}
+var (
+	typeStrings = [...]string{"int", "number", "string", "bool", "map", "array", "any", "alt"}
+
+	ValidRegistryName = regexp.MustCompile("^[a-z0-9](?:-?[a-z0-9])*$")
```

Comment:
should we allow starting with a number here?

Reply:
That's how it was in the `asserts/registry.go`, this just moves it. Is the registry name also being restricted to something similar to view names?

Reply:
I think so